inquire/src/ansi.rs

```diff
@@ -0,0 +1,92 @@
+use std::str::Chars;
+
+#[must_use]
+enum MatchResult<'a> {
+    Matched { remaining_chars: Chars<'a> },
+    NotMatched,
+}
+
+/// Match/consume the next ANSI escape code if it exists.
+/// The format is based on [this description](https://handwiki.org/wiki/ANSI_escape_code#CSI_(Control_Sequence_Introducer)_sequences)
+fn match_escape_code(chars: Chars<'_>) -> MatchResult<'_> {
+    let mut original_chars_iterator = chars;
+    let mut chars = original_chars_iterator.by_ref().peekable();
+
+    macro_rules! should_match {
+        ($c:expr, $pat:pat) => {
+            #[allow(unused_parens)]
+            match $c {
+                Some($pat) => {}
+                _ => return MatchResult::NotMatched,
+            }
+        };
+    }
+
+    // match \x1b[
+    should_match!(chars.next(), '\x1b');
+    should_match!(chars.next(), '[');
+
+    // match [0–9:;<=>?]*
+    while let Some('0'..='?') = chars.peek() {
+        chars.next();
+    }
+
+    // match [ !"#$%&'()*+,-./]
+    while let Some(' '..='/') = chars.peek() {
+        chars.next();
+    }
+
+    // match [@A–Z[\]^_`a–z{|}~]
+    should_match!(chars.next(), '@'..='~');
+
+    MatchResult::Matched {
+        remaining_chars: original_chars_iterator,
+    }
+}
+
+/// An iterator that strips ANSI escape codes from a string.
+///
+/// Generally constructed by calling [`strip_iter`].
+pub struct AnsiStripIter<'a> {
+    pub chars: Chars<'a>,
+}
+
+/// Constructs an iterator over the chars of the input string, stripping away ANSI escape codes.
```

Comment:
Let's make a Trait with this method `into_ansi_strip_iter()` and implement it for &str

Reply:
Good idea! I've implemented this in the last commit, but renamed it to `ansi_stripped_chars()` to more closely match `input.chars()` though, as this iterator is just a wrapper around `Chars`.